nilearn/glm/second_level/second_level.py

```diff
@@ -843,9 +843,24 @@ def non_parametric_inference(
     # Check design matrix and effect maps agree on number of rows
     _check_effect_maps(effect_maps, design_matrix)
 
+    # Obtain design matrix vars
+    var_names = design_matrix.columns.tolist()
```

Comment:
Is a `design_matrix` guaranteed? I think it's a mutually-exclusive parameter with `confounds`, so users could provide `confounds` and `design_matrix` should be None.

Reply:
Oh sorry, if this _is_ a bug, it's not one you've introduced, so you can ignore my comment.